File: susupay-backend/app/schemas/announcement.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class AnnouncementCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=120)
    body: str = Field(..., min_length=1, max_length=2000)
    is_pinned: bool = False


class AnnouncementUpdate(BaseModel):
    title: str | None = Field(None, min_length=1, max_length=120)
    body: str | None = Field(None, min_length=1, max_length=2000)
    is_pinned: bool | None = None


class AnnouncementResponse(BaseModel):
    id: uuid.UUID
    collector_id: uuid.UUID
    title: str
    body: str
    is_pinned: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class RatingCreate(BaseModel):
    score: int = Field(..., ge=1, le=5)
    comment: str | None = Field(None, max_length=500)


class RatingResponse(BaseModel):
    id: uuid.UUID
    collector_id: uuid.UUID
    client_id: uuid.UUID
    score: int
    comment: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class CollectorRatingSummary(BaseModel):
    average_score: float
    total_ratings: int
    ratings: list[RatingResponse]

File: susupay-frontend/src/types/announcement.ts
export interface Announcement {
  id: string;
  collector_id: string;
  title: string;
  body: string;
  is_pinned: boolean;
  created_at: string;
}

export interface AnnouncementCreate {
  title: string;
  body: string;
  is_pinned?: boolean;
}

export interface AnnouncementUpdate {
  title?: string;
  body?: string;
  is_pinned?: boolean;
}

export interface Rating {
  id: string;
  collector_id: string;
  client_id: string;
  score: number;
  comment: string | null;
  created_at: string;
}

export interface RatingCreate {
  score: number;
  comment?: string;
}

export interface CollectorRatingSummary {
  average_score: number;
  total_ratings: number;
  ratings: Rating[];
}

export interface ClientGroupOption {
  client_id: string;
  collector_name: string;
  group_invite_code: string;
}

export interface MultiGroupResponse {
  requires_group_selection: boolean;
  groups: ClientGroupOption[];
  selection_token: string;
}
